api/functions/decideStatus.js

```diff
@@ -0,0 +1,38 @@
+import logger from "../utils/logger.js";
+
+import { aggregateUserActivity } from "./aggregateUserActivity.js";
+import { decideScore } from "./decideScore.js";
+
+export const decideStatus = async (configTable, userId, userActivity) => {
+	try {
+		const aggregatedActivity = aggregateUserActivity(userId, userActivity);
+
+		// @todo updated version of decideScore should be replaced when it was mreged
+		if (aggregatedActivity.success) {
+			const score = decideScore({
+				messages: aggregatedActivity.countActivity.messagesCount,
+				reactions: aggregatedActivity.countActivity.reactionsCount,
+				messageWeight: configTable.message_weighting,
+				reactionWeight: configTable.reactions_weighting,
+			});
+
+			if (score < configTable.low_threshold) {
+				return { success: true, status: "inactive" };
+			} else if (score < configTable.medium_threshold) {
+				return { success: true, status: "low" };
+			} else if (score < configTable.high_threshold) {
+				return { success: true, status: "medium" };
+			} else {
+				return { success: true, status: "high" };
+			}
+		} else {
+			return {
+				success: false,
+				status: "activity has not been fetched correctly",
+			};
+		}
```

Comment:
For improved readability, we might benefit from using an early return pattern here to reduce nesting and make the code flow more clear.